code/modules/antagonists/mind_flayer/powers/flayer_mobility_powers.dm

```diff
@@ -0,0 +1,219 @@
+	marked_computer = null
+	should_recharge_after_cast = TRUE
+
+/datum/spell/flayer/computer_recall/AltClick(mob/user)
+	if(!marked_computer)
+		flayer.send_swarm_message("You do not current have a marked computer.")
+		return
+	flayer.send_swarm_message("Your current mark is [marked_computer].")
+
+/datum/spell/flayer/computer_recall/on_apply()
+	..()
+	cooldown_handler.recharge_duration -= 30 SECONDS
+
+/*
+ * Ok so this is slightly a stretch, but it hinders enemy mobility while not hindering the flayer
+ * It works like the hemomancer wall, creating up to 3 temporary walls
+ * Obtained for free in the Destroyer tree when reaching stage 3
+ */
+/datum/spell/flayer/techno_wall
+	name = "Crystalized Firewall"
+	desc = "Allows us to create a wall between two points. The wall is fragile and allows only ourselves to pass through."
+	base_cooldown = 60 SECONDS
+	action_icon_state = "pd_cablehop"
+	upgrade_info = "Double the health of the barrier by reinforcing it with ICE."
+	category = FLAYER_CATEGORY_DESTROYER
+	power_type = FLAYER_UNOBTAINABLE_POWER
+	base_cost = 100
+	current_cost = 100
+	max_level = 2
+	should_recharge_after_cast = FALSE
+	/// How big can we make our wall
+	var/max_walls = 3
+	/// Starting turf for the wall. Should be nulled after each cast or the cancelling of a cast
+	var/turf/start_turf
+
+/datum/spell/flayer/techno_wall/create_new_targeting()
+	var/datum/spell_targeting/click/T = new
+	T.allowed_type = /atom
+	T.try_auto_target = FALSE
+	return T
+
+/datum/spell/flayer/techno_wall/remove_ranged_ability(mob/user, msg)
+	. = ..()
+	if(msg) // this is only true if the user intentionally turned off the spell
+		start_turf = null
+		should_recharge_after_cast = FALSE
+
+/datum/spell/flayer/techno_wall/should_remove_click_intercept()
+	return start_turf
+
+/datum/spell/flayer/techno_wall/cast(list/targets, mob/user)
+	var/turf/target_turf = get_turf(targets[1])
+	if(target_turf == start_turf)
+		flayer.send_swarm_message("<span class='notice'>You deselect the targeted turf.</span>")
+		start_turf = null
+		should_recharge_after_cast = FALSE
+		return
+	if(!start_turf)
+		start_turf = target_turf
+		should_recharge_after_cast = TRUE
+		return
+	var/wall_count
+	for(var/turf/T as anything in get_line(target_turf, start_turf))
+		if(wall_count >= max_walls)
+			break
+		new /obj/structure/tech_barrier(T, 100 * level)
+		wall_count++
+
+	start_turf = null
+	should_recharge_after_cast = FALSE
+
+/obj/structure/tech_barrier
+	name = "crystalized firewall"
+	desc = "a strange structure of crystalised ... firewall? It's slowly melting away..."
+	max_integrity = 100
+	icon_state = "blood_barrier"
+	icon = 'icons/effects/vampire_effects.dmi'
+	density = TRUE
+	anchored = TRUE
+	opacity = FALSE
+	alpha = 200
+	var/upgraded_armor = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 50, RAD = 50, FIRE = 50, ACID = 50)
+	var/mutable_appearance/theme_icon
```

Comment:
Why is theme icon a variable on the type, does it get used outside of initiallize?